STS: Reorganize

Tirifto opened the pull request:
Summary:
- Reorganized the sections of the STS in a more sensible way.
- Added the "Friends and contacts" section, which explains that while Tox uses the term 'friend' internally in code, clients are recommended to call them 'contacts' for the user.
- Changed a link to "tox.im" into "tox.chat".

Arguments for:
- The new structure groups closely related subjects together and is easier to navigate and find things in.
- Calling 'friends' 'contacts' is considered to be a good change by many people and has been accepted in a separate PR for Tox-STS (which wasn't merged because it was poorly done, but the same is done better in this PR).
- "tox.im" is not used anymore and has been replaced by "tox.chat"

Arguments against:
- None that I am aware of.

Status quo:
- The content of STS is not organized well. Sections describing aspects of the same subject, or similar content, can be found in very different places. This makes the STS confusing and more difficult to navigate. The problem continues to get worse with each addition, since there's no organization pattern to follow.
- The STS says nothing about what contacts/friends should be called in the user interface.
- The outdated "tox.im" link prevails.

Main client maintainers, please share your votes and opinions.
@dvor @GrayHatter @Impyy @subliun @tux3

Commit:
STS: Reorganize

Summary:

* Reorganized the sections of the STS in a more sensible way.
* Added the "Friends and contacts" section, which explains that while Tox uses the term 'friend' internally in code, clients are recommended to call them 'contacts' for the user.
* Changed a link to "tox.im" into "tox.chat".

Arguments for:

* The new structure groups closely related subjects together and is easier to navigate and find things in.
* Calling 'friends' 'contacts' is considered to be a good change by many people and has been accepted in a separate PR for Tox-STS (which wasn't merged because it was poorly done, but the same is done better in this PR).
* "tox.im" is not used anymore and has been replaced by "tox.chat"

Arguments against:

* None that I am aware of.

Status quo:

* The content of STS is not organized well. Sections describing aspects of the same subject, or similar content, can be found in very different places. This makes the STS confusing and more difficult to navigate. The problem continues to get worse with each addition, since there's no organization pattern to follow.
* The STS says nothing about what contacts/friends should be called in the user interface.
* The outdated "tox.im" link prevails.

tsudoko commented:
I think this PR includes too many changed unrelated to each other, you might want to split it.

Tirifto commented:
> I think this PR includes too many changed unrelated to each other, you might want to split it.

I understand that and will do so, if it proves to be a problem. But I'd not not take the link change into consideration, since it's just a minor touch. That leaves two noteworthy unrelated changes, which I see as acceptable, so I hope it can work out.

Impyy commented:
What happened to the table of contents?
<img src="https://impy.me/i/9fb30f.png"/>

zetok commented:
Looks like a tad broken markdown;
instead of using ``### element``, ``* [element](#element)`` should be used

zetok commented:
@Impyy since it was fixed, is it now good?

Impyy commented:
@zetok I can't find any other oddities in this PR. So I approve.

GrayHatter commented on STS.md:
```diff
@@ -1,124 +1,94 @@
-- the client repeats step 3 until the bytes left is under ``TOX_MAX_MESSAGE_LENGTH``
-- The UI on the sender would show a new sent message for each broken section so they are aware.
-- The receiving client gets multiple separate messages like normal and displays them normally.
+The Tox ID and its associated friends list and save file is called a profile. Save files should use the ".tox" extension to denote that they are Tox save files; clients should recognize .tox save files, preferably via integration with the operating system. That way, merely by double clicking a .tox file, the client will import the file for use. Here, import means copy the file into the data direc ...
```
Tox data directory above*

GrayHatter commented:
:-1: with this reorg, a lot of text needs to be revised. There's a lot of redundant language that should be stripped. Additionally there's some parts that I don't believe should be in the STS. E.g. anything under discussion, or things like clients should do this, when 'this' isn't also give a recommended method, or things that have nothing to do with compatiblity.

The STS doesn't exist to make all clients the same, but to wake them all work well together.

zetok commented:
>The STS doesn't exist to make all clients the same, but to wake them all work well together.

Actually, STS is supposed to help clients look *all the same* - working together well is implied by the same look & feel.

GrayHatter commented:
That's stupid... Why even have other clients then?
On Sep 25, 2015 09:35, "zetok" <notifications@github.com> wrote:

> The STS doesn't exist to make all clients the same, but to wake them all
> work well together.
>
> Actually, STS is supposed to help clients look *all the same* - working
> together well is implied by the same look & feel.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Tox/Tox-STS/pull/76#issuecomment-143270748>.
>

zetok commented:
>Why even have other clients then?

Err. I always thought that you would know better than me answer to that question..

So, basically, qTox is not always the best solution - i.e. it's hard for people to compile it on distros that don't include Qt >=5.2 by default. It also depends on gcc >=4.8, which doesn't help.

That renders most of "older" distros a hard target for qTox.

Additionally, qTox doesn't really support XP - i.e. currently it works there without problems (from what I heard), but if it unexpectedly stopped to work there one day, this wouldn't be even considered to be a problem.

Now, µTox on the other hand, (AFAIK) uses small base of libs available on ~every distro, and given that it uses C99, there is no problem with compiler not being new enough.

Additionally, µTox supports XP - that didn't change, right?

====
Now, those IMHO are reasons good enough to have qTox and µTox - as for other clients, some people e.g. feel dissatisfied with something that e.g. doesn't use GTK for everything, etc.

GrayHatter commented:
Then why even use qTox?

I'm not arguing against other clients... just that clients SHOULD be different. 

The STS should define SOPs so that anyone who want's to make another completely different client, can still play nicely with the existing ones. No to make every client just appear as a clone to every other client.  Every Linux distro does things a little differently, but they still all use the same Kernel (mostly), and having all those choices is a good thing.

zetok commented:
>having all those choices is a good thing.

No, not really.
Choices are good when that is what people want - most of people want to get things done quickly without messing around, and done easily - having tons of options about which one first has to get some knowledge about is exactly opposite of "getting things done quickly and without messing around".

>The STS should define SOPs so that anyone who want's to make another completely different client, can still play nicely with the existing ones.

Not from PoV of "normal" user - if it differs greatly, then it's just hard for user to use what they want, and that makes differing clients **not** play nicely with each other.

In other words, you speak about about STS helping assure compatibility between clients, but sadly, you seem to think only about "technical" compatibility, and forget about the most important compatibility from PoV of user: interface compatibility. Clients that don't have that.. how could they be said to be compatible?

>No to make every client just appear as a clone to every other client.

Dunno about that - if you say that being a "clone" is a bad thing, why not simply improve things that are bad in all clients, so that there would be a **good interface shared among clients**, and improvements would be also shared among them?

Realistically though, I have doubts whether that would be doable in every single aspect of the UI. Doesn't mean that one shouldn't try `:)`

====
Now, that's what STS should be for.

But thing is, there will be (already is) a need from some people to create clients that *won't* have interface compatibility, since that's the main point why those clients are being made.

Arising question is how STS should be adapted - perhaps it should be extended to say what is compatible UI is, and say that it's **really recommended**, although not *mandatory* to get things to work properly.



====
>Then why even use qTox?
>
>I'm not arguing against other clients... just that clients SHOULD be different. 

:D

You know - qTox is simply better :P.
Reason why it's better is because UI is being improved based other clients as well - it takes good parts, and does them "the right way". Now, it's not like doing things "the right way" is impossible in other clients..
But lemme tell an anecdote.

Once upon a time, @notsecure implemented screenshots in µTox. And he knew that it was good, for people praised his name for it.
But then, came a day when appeared voices of people dissatisfied by the way functionality was done, for it lacked in UI.
Among those voices, I was. And for what I witnessed, functionality could be improved to be better, so that people could be even happier using it.
Thus I spoke, and @notsecure after hearing my reasoning dismissed my words, viewing improvement as unnecessary, for UI he made was "good enough".
Greatly disappointed about this, I've turned to the other client, qTox was its name.
And so my words were heard again. Not to be dismissed, but kept in mind, for @tux3 in his wisdom said "convince me that people want this".
Time has passed and more people were convinced of the better way of screenshots.
And so day came that screenshots were implemented and the great merge happened, for the people to enjoy this great feature.
Since that day, qTox users enjoyed better screenshots.


----
Now, I would like to point out that if not for outright dismissal of UI changes that could improve features, µTox would also have had a good way of doing screenshots.
Thing is, attitude.
@notsecure viewed UI as "good enough", and thus he **refused to change it to improve it**.

I.e. the only reason why µTox differs in UI from qTox is because it was viewed as "good enough", and as such started to lack, since qTox did try to improve UI, rather than outright refusing.

Dunno how you @GrayHatter are going about things currently - hopefully that kind of thing is no longer the case `:)`

And if so, then I don't see why UI shouldn't be shared and mutually improved, so that the best user facing interface could be used by both qTox and µTox `:)`

Tirifto commented:
@GrayHatter I fixed the word and will try to do the same for any errors to be found in the future.

Concerning the actual contents of STS, I'm not sure if modifying those belongs in this PR. It is true that I have added a section. It adds content which, to my knowledge, is both welcomed by community and helpful for user experience, since varying terminology is confusing.

I have little experience with contributions, so I'm not sure what all should I change here and how. But if others decide on that, I will commit the changes. But as things are now, this is mainly a reorganization. If there should be major changes to the content that would take a long time to discuss, perhaps that should be done in a different PR, in order to not hold other changes back.

–––––––––––––

@zetok I'm not sure if I understand your opinion 100% well, but here is mine: Clients should be similar, but not the same. @GrayHatter makes a good point with GNU/Linux distributions. There's a lot of them and they can be very different. But so can be the users' tastes in interfaces.

I think good design is important, but I also think there shouldn't be only one, because it won't suit all the people. I see no problems in clients having somewhat different interfaces, because that way Tox could appeal to more users. Sure, some design would be better than the others, and that one can be advertised and recommended more. That said, while I don't see all clients looking exactly the same as good, putting some parts of the design into STS could be good.

If you look at GNOME Shell, Unity and Cinnamon, you can see that they are different. But you can also use them without much trouble, because all three of them share common elements: There's a panel with time and date and battery/wifi/sound indicators. There's a button you can click to bring up a menu, that- while different in each of these -will let you run programs quickly. You get windows that have title and close button at the top, and you can resize them by dragging corners.

Summed up: I think it wouldn't be bad to add some interface guidelines to the STS. That way we can create familiar look and good design to be shared across clients. But I don't think it should be a complete interface design, but a vague one, or just general reccomendations on different parts of the UI.

I'm using GNOME Shell at the moment, and I appreciate how gTox tries to integrate into GNOME and make use of its features. It wouldn't be able to do what it does now if it followed a design that didn't take GNOME into consideration.

Of course, the STS is not a law. If someone really wants to make their own UI design, they'll make it whether STS even mentions UI or not. But if it is to mention UI, I think this is the best way it can do it.

tux3 commented:
Looks OK to me. :+1:

subliun commented:
Looks good.
